### ISSUE TITLE
Unable to click i3 tray

### ISSUE THREAD

**Issue by aadilshabier** [polybar/polybar#2460]
## Checklist

* [x] I have read the appropriate section in the [contributing
  guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
* [x] I believe this issue is a problem with polybar itself and not a misconfiguration on my part.
* [x] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [x] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.

## Describe the bug
On restarting i3-gaps in place with Mod+Shift+r, the tray icons become unclickable.

**Expected behavior:**
The tray icons should be clickable after restarting i3-gaps in place.

**Actual behavior:**
The tray icons only become clickable once polybar is relaunched with the launch script:
```bash
#!/usr/bin/env sh
killall -q polybar
echo "---" | tee -a /tmp/polybar1.log /tmp/polybar2.log
polybar top 2>&1 | tee -a /tmp/polybar1.log & disown
```

**Was it working before?**
It hasn't worked on this machine since I started using polybar.

## To Reproduce
```dosini
[bar/top]
modules-left = i3
modules-center = mpd
modules-right = xbacklight xkeyboard filesystem memory cpu temperature date battery powermenu
tray-position = right
tray-padding = 2
tray-background = #80000000
wm-restack = i3
scroll-up = #i3.prev
scroll-down = #i3.next
cursor-click = pointer
cursor-scroll = ns-resize

[module/i3]
type = internal/i3
format = <label-state> <label-mode>
index-sort = true
wrapping-scroll = false
pin-workspaces = true
label-mode-padding = 2
```
1. Star polybar with the launch script given above.
2. Restart i3 in place

## Polybar Log
During first launch
```
polybar|notice:  Parsing config file: /home/aadil/.config/polybar/config
# Loaded fonts
polybar|warn:  Ignoring restack of i3 window (not needed when `override-redirect = false`)
polybar|warn:  Systray selection already managed (window=0x3c0000a)
polybar|error: tray: Failed to put tray above 0x3e00003 in the stack (XCB_MATCH (8))
```
After restarting i3
```
polybar|error: module/i3: Failed to reconnect socket (reason: Failed to connect to i3: errno 2 (No such file or directory))
polybar|error: module/i3: Failed to reconnect socket (reason: Failed to connect to i3: errno 2 (No such file or directory))
polybar|error: module/i3: Failed to reconnect socket (reason: Failed to connect to i3: errno 2 (No such file or directory))
polybar|error: module/i3: Failed to reconnect socket (reason: Failed to connect to i3: errno 2 (No such file or directory))
```
## Environment:
* WM: i3 version 4.19.1 (2021-02-01)
* Distro: Arch Linux x86_64 5.12.14-arch1-1
* Output of `polybar -vvv`:

```
polybar 3.5.6

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -Wall -Wextra -Wpedantic -O3 -DNDEBUG
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```

## Additional context
Screenshot when unable to click (cursor scrolling instead of selecting)
![polybarbug](https://user-images.githubusercontent.com/64919407/125189531-30e04100-e256-11eb-95cb-49ff11e2181d.jpeg)

**Comment by patrick96** [comment]
I can kind of reproduce this. But for me, after restarting, the tray isn't visible at all. But I believe the underlying issue is the same as in #425.

The tray is a separate window and in i3 (without `override-redirect`), it's almost impossible to correctly position the tray correctly on the window stack relative to the bar window because i3 sometimes moves them around the window stack.

**Comment by patrick96** [comment]
This should be fixed by #2609 if you are using the new tray module.
